File: src/Blog/Blog.Blazor/Data/Entities/ProjectTranslation.cs
using Blog.Blazor.Data;

namespace Blog.Blazor.Data.Entities;

public sealed class ProjectTranslation
{
    public int Id { get; set; }
    public int ProjectId { get; set; }
    public string Locale { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Purpose { get; set; } = string.Empty;
    public string? Problem { get; set; }
    public string? CurrentFocus { get; set; }
    public string? Status { get; set; }

    [JsonOrEmpty]
    public string? Metrics { get; set; }

    public string? Body { get; set; }

    [JsonOrEmpty]
    public string? Seo { get; set; }

    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public Project? Project { get; set; }
}

File: src/laravel/app/Models/ProjectTranslation.php
<?php

namespace App\Models;

use App\Models\Concerns\Auditable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class ProjectTranslation extends Model
{
    use Auditable, HasFactory;

    protected $fillable = ['project_id', 'locale', 'name', 'purpose', 'problem', 'current_focus', 'status', 'metrics', 'body', 'seo'];

    protected $casts = ['metrics' => 'array', 'seo' => 'array'];

    /**
     * @return BelongsTo<Project, $this>
     */
    public function project(): BelongsTo
    {
        return $this->belongsTo(Project::class);
    }
}
